### PR TITLE
[doc][yba][2024.1.2] Export database audit logs

### PR DESCRIPTION
**Pull request by ddhodge**
Export audit logs
PLAT-14051

@netlify /preview/yugabyte-platform/alerts-monitoring/universe-logging/

### CODE REVIEW

File: docs/content/preview/yugabyte-platform/alerts-monitoring/anywhere-export-configuration.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+- [Datadog](https://docs.datadoghq.com/)
+- [Splunk](https://www.splunk.com/en_us/solutions/opentelemetry.html)
+- [AWS CloudWatch](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/WhatIsCloudWatch.html)
+- [GCP Cloud Logging](https://cloud.google.com/logging/)
+
+For information on how to export logs from a universe using an export configuration, refer to [Export logs](../universe-logging/).
+
+## Prerequisites
+
+Export configuration is {{<badge/ea>}}. To enable export configuration management, set the **Enable DB Audit Logging** Global Configuration option (config key `yb.universe.audit_logging_enabled`) to true. Refer to [Manage runtime configuration settings](../../../administer-yugabyte-platform/manage-runtime-config/). Note that only a Super Admin user can modify Global configuration settings. The flag can't be turned off if audit logging is enabled on a universe.
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/alerts-monitoring/anywhere-export-configuration.md:32]:
```suggestion
Export configuration is {{<badge/ea>}}. To enable export configuration management, set the **Enable DB Audit Logging** Global Configuration option (config key `yb.universe.audit_logging_enabled`) to true. Refer to [Manage runtime configuration settings](../../administer-yugabyte-platform/manage-runtime-config/). Note that only a Super Admin user can modify Global configuration settings. The flag can't be turned off if audit logging is enabled on a universe.
```

File: docs/content/preview/yugabyte-platform/alerts-monitoring/anywhere-export-configuration.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+## Configure integrations
+
+Create and manage export configurations on the **Integrations > Log** page.
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/alerts-monitoring/anywhere-export-configuration.md:36]: I can't see "Logs" on portal.dev. Assuming thats expected as its on 2.23.x

File: docs/content/preview/yugabyte-platform/alerts-monitoring/anywhere-export-configuration.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+![Export configurations](/images/yp/export-configurations.png)
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/alerts-monitoring/anywhere-export-configuration.md:38]: Image is missing.

**Review by aishwarya24 — COMMENTED** [review]
some comments.

File: docs/content/preview/yugabyte-platform/alerts-monitoring/anywhere-export-configuration.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    identifier: anywhere-export-configurations
+    parent: alerts-monitoring
+    weight: 80
+type: docs
+---
+
+You can export universe logs to third-party tools for analysis and customization. Exporting data is a two-stage process:
+
+1. Create an export configuration. A configuration defines the sign in credentials and settings for the tool that you want to export to.
+1. Use the configuration to export data from a universe. While the connection is active, metrics or logs are automatically streamed to the tool.
```
Comment (Sahith02) [inline docs/content/preview/yugabyte-platform/alerts-monitoring/anywhere-export-configuration.md:19]: Change "data" to "audit logs"?
Change "metrics or logs" to just "logs"

File: docs/content/preview/yugabyte-platform/alerts-monitoring/anywhere-export-configuration.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+1. Optionally, enter the Source, Source Type, and Index.
+1. Click **Validate and Create Configuration**.
+
+  {{% /tab %}}
+
+  {{% tab header="AWS" lang="aws" %}}
+
+The AWS CloudWatch export configuration requires the following:
+
+- Access Key ID and Secret Access Key for the AWS Service Account
```
Comment (Sahith02) [inline docs/content/preview/yugabyte-platform/alerts-monitoring/anywhere-export-configuration.md:87]: Should we mention the other requirements such as Log group, Log stream, and region here?

File: docs/content/preview/yugabyte-platform/alerts-monitoring/universe-logging.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+| :----- | :----- | :------ |
+| pgaudit.log_catalog | Log statements for the PostgreSQL system catalog relations in `pg_catalog`. These system catalog tables record system (as opposed to user) activity, such as metadata lookups and from third-party tools performing lookups.<br>These statements aren't required for typical auditing and you can disable this option to reduce noise in the log. | ON |
+| pgaudit.log_client | Enable this option to echo log messages directly to clients such as [ysqlsh](../../../admin/ysqlsh/) and psql. Log messages are printed directly to the shell, which can be helpful for debugging.<br>When enabled, you can set the level of logs that are output using `pgaudit.log_level`. | OFF |
+| pgaudit.log_level | Sets the [severity level](https://www.postgresql.org/docs/11/runtime-config-logging.html#RUNTIME-CONFIG-SEVERITY-LEVELS) of logs written to clients when `pgaudit.log_client` is on. Use this setting for debugging and testing.<br>Values: DEBUG1 .. DEBUG5, INFO, NOTICE, WARNING, LOG.<br>ERROR, FATAL, and PANIC are not allowed.<br>`pgaudit.log_level` only applies when `pgaudit.log_client` is on; otherwise the default LOG level is used. | LOG |
```
Comment (Sahith02) [inline docs/content/preview/yugabyte-platform/alerts-monitoring/universe-logging.md:73]: Also need to mention setting the log_min_messages gflag for this pgaudit.log_level param to work properly. This is mentioned in the doc content file.

**Review by Sahith02 — COMMENTED** [review]
Added a few comments, rest looks good to me. Please get it reviewed by the PM also once (Ranga).

File: docs/content/preview/yugabyte-platform/alerts-monitoring/universe-logging.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
+---
+title: Export PostgreSQL logs from YugabyteDB Anywhere universes
```
Comment (rangasayeelc-yb) [inline docs/content/preview/yugabyte-platform/alerts-monitoring/universe-logging.md:2]: "PostgreSQL logs" -> shouldn't we say YSQL audit logs? Same question for the subtitle here in YBM / Aeon https://deploy-preview-23605--infallible-bardeen-164bc9.netlify.app/preview/yugabyte-cloud/cloud-monitor/logging-export/

File: docs/content/preview/yugabyte-platform/alerts-monitoring/universe-logging.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+
+<!-- Database query logging. This is the standard [PostgreSQL logging](https://www.postgresql.org/docs/11/runtime-config-logging.html) facility. Using these settings you can log query statements and errors.-->
+- Database audit logging. Using the [PostgreSQL Audit Extension](https://www.pgaudit.org/#) ([pgaudit](https://github.com/pgaudit/pgaudit/blob/1.3.2/README.md)), the audit log provides the exact database transactions, which is a compliance requirement for government, financial, or ISO certifications.
+
+Note that YugabyteDB is based on PostgreSQL 11 and uses pgaudit v1.3.2.
+
+## Prerequisites
+
+The Audit log export feature is {{<badge/ea>}}. To enable the feature in YugabyteDB Anywhere, set the **Enable DB Audit Logging** Global Configuration option (config key `yb.universe.audit_logging_enabled`) to true. Refer to [Manage runtime configuration settings](../../../administer-yugabyte-platform/manage-runtime-config/). Note that only a Super Admin user can modify Global configuration settings. The flag can't be turned off if audit logging is enabled on a universe.
```
Comment (rangasayeelc-yb) [inline docs/content/preview/yugabyte-platform/alerts-monitoring/universe-logging.md]: @Sahith02 if the feature is behind a UI / API/ CLI feature flag, why do we need to keep it disabled this way as well?

File: docs/content/preview/yugabyte-platform/alerts-monitoring/universe-logging.md [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+
+To export logs, you need to first create an export configuration. A configuration defines the sign in credentials and settings for the tool that you want to export your logs to. Refer to [Manage export configurations](../anywhere-export-configuration).
+
+## Limitations
+
+- Log export is only supported for cloud-based universes and automatically provisioned on-premises universes. Manually provisioned universes and Kubernetes universes are not supported.
+- Log export uses a PostgreSQL extension and is only compatible with YSQL.
```
Comment (rangasayeelc-yb) [inline docs/content/preview/yugabyte-platform/alerts-monitoring/universe-logging.md]: rephrase second limitation to, "Only YSQL database audit logs can be exported, not YCQL"

File: docs/content/preview/yugabyte-platform/alerts-monitoring/universe-logging.md [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+- There are also no alerts or metrics currently available for this feature.
```
Comment (rangasayeelc-yb) [inline docs/content/preview/yugabyte-platform/alerts-monitoring/universe-logging.md]: I don't think we need to say this as it is orthogonal to log export

File: docs/content/preview/yugabyte-platform/alerts-monitoring/universe-logging.md [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+    identifier: universe-logging
+    parent: alerts-monitoring
+    weight: 70
+type: docs
+---
+
+Export YSQL database logs to third-party tools for security monitoring, to build operations and health dashboards, troubleshooting, and more. You can export the following types of logs:
+
+<!-- Database query logging. This is the standard [PostgreSQL logging](https://www.postgresql.org/docs/11/runtime-config-logging.html) facility. Using these settings you can log query statements and errors.-->
+- Database audit logging. Using the [PostgreSQL Audit Extension](https://www.pgaudit.org/#) ([pgaudit](https://github.com/pgaudit/pgaudit/blob/1.3.2/README.md)), the audit log provides the exact database transactions, which is a compliance requirement for government, financial, or ISO certifications.
```
Comment (rangasayeelc-yb) [inline docs/content/preview/yugabyte-platform/alerts-monitoring/universe-logging.md:18]: @ddhodge @Sahith02 
do we need to mention the pgaudit version as users cannot change it and the supported logging parameters are documented in this article? what do we do for other extensions / plugins that we support?

**Review by rangasayeelc-yb — CHANGES_REQUESTED** [review]

**Pull request merged**